### PR TITLE
HPCC-21944 Folding function calls that return real values may fail

### PR DESCRIPTION
**Pull request by richardkchapman**
Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [x] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:
Using Ubuntu 1804 and the changes in HPCC-21879, zerodivide1.ecl test was failing prior to this fix.

Note that the changes in HPCC21879 are not directly related to this issue, they just happened to give the gcc compiler just that little bit more ability to optimize generated code and mis-optimize this bit.

### CODE REVIEW

**Comment by richardkchapman** [comment]
@ghalliday 

Note that I have only tested the x64 version - I don't know if the arm / x86 versions were affected or whether the fix here for them isright (though so long as it compiles, it should be at least no worse).

File: ecl/hql/hqlfold.cpp [hunk 1]
```diff
@@ -1230,15 +1230,15 @@ IValue * doFoldExternalCall(IHqlExpression* expr, unsigned foldOptions, ITemplat
             {
                 __asm__  __volatile__(
                     "fsts  s0,[%[fresult]] \n\t"
-                    :
+                    :"=m"floatresult)
```
Comment (ghalliday) [inline ecl/hql/hqlfold.cpp]: missing (

**Review by ghalliday — COMMENTED** [review]
@richardkchapman one problem.  Who can test the different variants?

**Comment by richardkchapman** [comment]
@ghalliday Fixed the missing (, and tested the ARM variants as best I can using https://godbolt.org/

**Comment by ghalliday** [comment]
change looks good.  Should it target an earlier version though.

**Pull request merged**